pkg/util/strategicpatch/patch.go

```diff
@@ -167,6 +213,12 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	if t.Kind() != reflect.Map {
 		// Maybe in the future add a "concat" mode that doesn't
 		// uniqify.
+		if !reflect.DeepEqual(original, patch) {
+			dw.WriteRemoval(fmt.Sprintf("%v", original))
+			dw.WriteAddition(fmt.Sprintf("%v", patch))
+		} else {
+			dw.WriteSame(fmt.Sprintf("%v", original))
+		}
```

Comment:
Hi @kbeecher, you are right, it should be 
  a
- b
- c
+ d,

Currently the code will produce, 
- a
- b
- c
+ a
+ d,
right? So it still needs some work.

I was reviewing this PR with the impression that we are generating diff between `original` and `original+patch`, sorry for the confusion. I need to review the PR again.